packages/backend/src/tools/updateDiffHistory.ts

```diff
@@ -176,11 +230,21 @@ function generateDiffHistoryMarkdown(
     )
     result.push('')
   }
+
   result.push('## Watched changes')
   result.push('')
   result.push(discoveryDiffToMarkdown(diffs))
   result.push('')
 
+  if (codeDiff !== '') {
```

Comment:
I don't like this compare against an empty string, assumes that we know what is going on outside of this function. Maybe it can be optional just like the `description`?